internal/controllers/utils/utils.go

```diff
@@ -771,6 +772,68 @@ func CheckClusterLabelsForPolicies(
 	return nil
 }
 
+// FindClusterInstanceImmutableFieldUpdates identifies updates made to immutable fields
+// in the ClusterInstance spec. It returns a list of paths for the updated fields that
+// are considered immutable and should not be modified.
+func FindClusterInstanceImmutableFieldUpdates(
+	oldClusterInstance, newClusterInstance *unstructured.Unstructured) ([]string, error) {
+
+	// Non-immutable ClusterInstance fields at the cluster-level
+	allowedClusterFields := []string{
```

Comment:
I would suggest pulling these 2 lists out into `constants.go` so that they were separate from the code.

Reply:
I didn’t put it in constants.go initially because I thought only `const` types should be added there, but it looks like we already have `var` in that file as well. I’ll update it.